src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService_FindDeclaration.cs

```diff
@@ -164,16 +172,20 @@ class NestedType
                 }
             }
 
-            // Generate into any decl we can find.
+            // If there is a declaration in a non auto-generated file, prefer it.
+            Func<Document,bool> isGeneratedCode = 
+                solution.Workspace.Services.GetService<IGeneratedCodeRecognitionService>().IsGeneratedCode;
```

Comment:
Partially generated or fully, the [service correctly tags *designer](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/GeneratedCodeRecognition/GeneratedCodeRecognitionServiceFactory.cs,ef3599fb042e3706,references) as generated by tool, which serves my purpose :smile: 

Also curious, what motivates adding user code into a file that is touched by the designer, when  its implemented as a partial type in more than 1 locations afterall ?